test/variable_test.cpp

```diff
@@ -956,3 +956,39 @@ TEST(VariableSlice, slice_binary_operations) {
   EXPECT_TRUE(equals(difference.get<const Data::Value>(), {-1, -1}));
   EXPECT_TRUE(equals(product.get<const Data::Value>(), {2, 12}));
 }
+
+TEST(Variable, reshape) {
+  Variable var(Data::Value{}, {{Dim::X, 2}, {Dim::Y, 3}}, {1, 2, 3, 4, 5, 6});
+  auto view = var.reshape({Dim::Row, 6});
+  ASSERT_EQ(view.size(), 6);
+  ASSERT_EQ(view.dimensions(), Dimensions({Dim::Row, 6}));
+  EXPECT_TRUE(equals(view.get<const Data::Value>(), {1, 2, 3, 4, 5, 6}));
+
+  auto view2 = var.reshape({{Dim::Row, 3}, {Dim::Z, 2}});
+  ASSERT_EQ(view2.size(), 6);
+  ASSERT_EQ(view2.dimensions(), Dimensions({{Dim::Row, 3}, {Dim::Z, 2}}));
+  EXPECT_TRUE(equals(view2.get<const Data::Value>(), {1, 2, 3, 4, 5, 6}));
+}
+
+TEST(Variable, reshape_fail) {
+  Variable var(Data::Value{}, {{Dim::X, 2}, {Dim::Y, 3}}, {1, 2, 3, 4, 5, 6});
+  EXPECT_THROW_MSG(var.reshape({Dim::Row, 5}), std::runtime_error,
```

Comment:
`std::runtime_error` -> `std::runtime_error&` unless macro does this for you already

Reply:
The macro does this.